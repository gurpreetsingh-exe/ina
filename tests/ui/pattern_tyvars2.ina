// COMPILE_FAIL

extern mod core;
using core::prelude::*;

using Foo::*;
type Foo =
    | T1(Option[isize])
    | T2

fn main() {
    match T2 {
        T1(Some(x)) => {
            let x2: i32 = x;
            x2
        },
        T1(None) => 0,
        T2 => 0,
    };
}
