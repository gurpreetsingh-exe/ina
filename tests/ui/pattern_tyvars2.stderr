error: mismatch types
    ╭─[tests/ui/pattern_tyvars2.ina:14:27]
    │
 14 │             let x2: i32 = x;
    │                           ┬
    │                           ╰─ expected `i32`, found `isize`
    ∙
────╯
